### PR TITLE
Added support for using an ES6 default exported function for configuration

### PR DESCRIPTION
**Pull request by johanneslumpe**

### CODE REVIEW

**Comment by johanneslumpe** [comment]
Solves https://github.com/webpack/webpack/issues/2105

File: bin/convert-argv.js [hunk 1]
```diff
@@ -94,7 +94,9 @@ module.exports = function(optimist, argv, convertOptions) {
 		configFileLoaded = true;
 	}
 
-	if(typeof options === "function") {
+	var isES6DefaultExportedFunc = (typeof options === "object" && typeof options["default"] === "function");
```
Comment (sokra) [inline bin/convert-argv.js]: Could you add a check for `null`?

``` js
var isES6DefaultExportedFunc = (options && typeof options === "object" && ...
```

Maybe also a few lines below in my code. I've done the same mistake ;)

Comment (johanneslumpe) [inline bin/convert-argv.js]: Haha ;) Will add that later today!

File: bin/convert-argv.js [hunk 2]
```diff
@@ -94,14 +94,19 @@ module.exports = function(optimist, argv, convertOptions) {
 		configFileLoaded = true;
 	}
 
-	if(typeof options === "function") {
+	var isES6DefaultExportedFunc = (
+		typeof options === "object" && options !== null && typeof options["default"] === "function"
```
Comment (StephanBijzitter) [inline bin/convert-argv.js]: `options["default"]` should be rewritten as `options.default`, it's just cleaner that way

Comment (johanneslumpe) [inline bin/convert-argv.js]: Updated

**Comment by johanneslumpe** [comment]
@sokra Is anything missing here?

**Comment by sokra** [comment]
Thanks

**Pull request merged**